src/api/expenses.controller.js

```diff
@@ -0,0 +1,89 @@
+const { expensesService } = require('../services/expenses.service.js');
+const { usersService } = require('../services/users.service.js');
+
+const getExpenses = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const categoryList = categories ? categories.split(',') : [];
+
+  const expenses = expensesService.getExpenses({
+    userId,
+    categoryList,
+    from,
+    to,
+  });
+
+  res.json(expenses);
+};
+
+const createExpense = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId || !spentAt || !title || !amount || !category || !note) {
+    return res.sendStatus(400);
+  }
+
+  const user = usersService.getUserById(userId);
+
+  if (!user) {
+    return res.sendStatus(400);
+  }
+
+  const expense = expensesService.createExpense({
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  });
+
+  res.status(201).json(expense);
+};
+
+const getExpenseById = (req, res) => {
+  const expense = expensesService.getExpenseById(+req.params.id);
+
+  if (!expense) {
+    return res.sendStatus(404);
+  }
+
+  res.json(expense);
+};
+
+const deleteExpenseById = (req, res) => {
+  const expense = expensesService.deleteExpenseById(+req.params.id);
```

Comment:
The `deleteExpenseById` function should ensure that the `expensesService.deleteExpenseById` method correctly returns a value indicating whether the deletion was successful. If it returns `undefined` or `null`, the current check will work, but if it returns a boolean, you might need to adjust the logic.